happypose_msgs/happypose_msgs_py/symmetries.py

```diff
@@ -0,0 +1,98 @@
+from copy import copy
+import numpy as np
+import numpy.typing as npt
+import transforms3d
+from typing import List, Union
+
+from geometry_msgs.msg import Transform, Vector3, Quaternion
+
+from happypose_msgs.msg import ObjectSymmetries
+
+
+def discretize_symmetries(
+    object_symmetries: ObjectSymmetries,
+    n_symmetries_continuous: int = 8,
+    return_ros_msg: bool = False,
+) -> Union[npt.NDArray[np.float64], List[Transform]]:
+    """Converts discrete and continuous symmetries to a list of discrete symmetries.
+
+    :param object_symmetries: ROS message containing symmetries of a given object.
+    :type object_symmetries: happypose_msgs.msg.ObjectSymmetries
+    :param n_symmetries_continuous: Number of segments to discretize continuous symmetries.
+    :type n_symmetries_continuous: int
+    :param return_ros_msg: Whether to return ROS message or numpy array
+        with 4x4 matrices, defaults to False.
+    :type return_ros_msg: bool, optional
+    :return: If ``return_ros_msg`` is False returns array of a shape (n, 4, 4) with ``n``
+        SE3 transformation matrices representing symmetries.
+        Otherwise list of ROS Transform messages.
+    :rtype: Union[npt.NDArray[np.float64], List[geometry_msgs.msg.Transform]]
+    """
+
+    # If there are no continuous symmetries and ROS message is expected skip computations
+    if return_ros_msg and len(object_symmetries.symmetries_continuous) == 0:
+        return copy(object_symmetries.symmetries_discrete)
+
+    n_con = len(object_symmetries.symmetries_continuous) * n_symmetries_continuous
+    n_disc = len(object_symmetries.symmetries_discrete)
+    n_mix = n_con * n_disc
+
+    # Preallocate memory for results
+    out = np.zeros((n_con + n_disc + n_mix, 4, 4))
+
+    # Precompute steps of rotations
+    rot_base = 2.0 * np.pi / n_symmetries_continuous
```

Comment:
`np.linspace` will include `2 pi` in the resulting list, creating redundant value.
Code below compares results:
```python
n_symmetries_continuous = 6
rot_base = 2.0 * np.pi / n_symmetries_continuous

lin_space = np.linspace(0.0, 2.0 * np.pi, n_symmetries_continuous)
manual = np.array([rot_base * j for j in range(n_symmetries_continuous)])

print(f"Linspace: {np.rad2deg(lin_space)}")
print(f"Manual:   {np.rad2deg(manual)}")
```
With an output
```
Linspace: [  0.  72. 144. 216. 288. 360.]
Manual:   [  0.  60. 120. 180. 240. 300.]
```
This can be equal if line
```python
lin_space = np.linspace(0.0, 2.0 * np.pi, n_symmetries_continuous)
```
was replaced with
```python
lin_space = np.linspace(0.0, 2.0 * np.pi, n_symmetries_continuous+1)[:-1]
```
But to be fair, this makes it even less readable.

I would honestly leave it as it is

Reply:
You can do this:

```
np.linspace(0,10, 10, endpoint=False)
>> array([0., 1., 2., 3., 4., 5., 6., 7., 8., 9.])
```

Reply:
Ok. I ended up with the linspace version as requested